ibc-clients/cw-context/src/handlers.rs

```diff
@@ -178,26 +182,28 @@ where
                 );
 
                 let consensus_state = self.consensus_state(&client_cons_state_path)?;
+                let timestamp = consensus_state.timestamp().nanoseconds();
 
-                QueryResponse::success().timestamp(consensus_state.timestamp().nanoseconds())
+                to_json_binary(&TimestampAtHeightResponse { timestamp })
             }
             QueryMsg::VerifyClientMessage(msg) => {
                 let msg = VerifyClientMessageMsg::try_from(msg)?;
 
-                client_state.verify_client_message(self, &client_id, msg.client_message)?;
+                let is_valid = client_state
+                    .verify_client_message(self, &client_id, msg.client_message)
+                    .is_ok();
 
-                QueryResponse::success()
+                to_json_binary(&VerifyClientMessageResponse { is_valid })
             }
             QueryMsg::CheckForMisbehaviour(msg) => {
                 let msg = CheckForMisbehaviourMsg::try_from(msg)?;
 
-                let result =
+                let found_misbehaviour =
                     client_state.check_for_misbehaviour(self, &client_id, msg.client_message)?;
 
-                QueryResponse::success().misbehaviour(result)
+                to_json_binary(&CheckForMisbehaviourResponse { found_misbehaviour })
             }
-        };
-
-        Ok(to_json_binary(&resp)?)
```

Comment:
I've moved `to_json_binary` inside the match statement because otherwise each match arm returns a different struct.